File: backend/src/main/java/com/zychen/backend/service/WordbookService.java
package com.zychen.backend.service;

import com.zychen.backend.dto.response.WordbookPage;

import java.util.Map;

public interface WordbookService {

    /**
     * 加入生词本：无记录则插入 status=2；已有记录则更新为 status=2
     */
    void addToWordbook(Long userId, Long wordId);

    /**
     * 移出生词本：将 status 置为 0（学习中）
     */
    void removeFromWordbook(Long userId, Long wordId);

    /**
     * 分页生词本列表（关联 word 详情）
     */
    WordbookPage getWordbookList(Long userId, int page, int size);

    /**
     * 生词本词条总数（status=2）
     */
    int getWordbookCount(Long userId);

    /**
     * 获取 AI 生成例句（同步调用）
     *
     * @return data: { word, examples（原例句）, aiExample（AI例句） }
     */
    Map<String, Object> getAIContent(Long userId, Long wordId);
}

File: frontend/lib/services/wordbook_service.dart
// lib/services/wordbook_service.dart
import '../core/network/api_client.dart';
import '../core/constants/api_constants.dart';
import '../models/wordbook_model.dart';

class WordbookService {
  final ApiClient _apiClient = ApiClient();

  Future<List<WordbookWord>> getWordbookList({
    int page = 1,
    int size = 20,
  }) async {
    final response = await _apiClient.get(
      ApiConstants.wordbookList,
      queryParams: {'page': page, 'size': size},
    );

    final data = response['data'] as Map<String, dynamic>? ?? {};
    final list = (data['content'] ?? data['list'] ?? <dynamic>[]) as List<dynamic>;
    return list
        .map((e) => WordbookWord.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  Future<void> addToWordbook(int wordId) async {
    await _apiClient.post('${ApiConstants.wordbookAdd}/$wordId');
  }

  Future<void> removeFromWordbook(int wordId) async {
    await _apiClient.delete('${ApiConstants.wordbookRemove}/$wordId');
  }

  Future<AIContentResponse> getAIContent(int wordId) async {
    final response = await _apiClient.get('${ApiConstants.wordbookAi}/$wordId');
    return AIContentResponse.fromJson(response);
  }
}
